pilot/pkg/kube/inject/app_probe_test.go

```diff
@@ -43,3 +43,24 @@ func TestFindSidecar(t *testing.T) {
 		}
 	}
 }
+
+func TestShouldRewriteAppProbers(t *testing.T) {
+	for _, tc := range []struct {
+		name                 string
+		sidecarInjectionSpec SidecarInjectionSpec
+		annotations          map[string]string
+		expected             bool
+	}{
+		{"RewriteAppHTTPProbe-not-set", SidecarInjectionSpec{RewriteAppHTTPProbe: false}, map[string]string{}, false},
+		{"RewriteAppHTTPProbe-set-in-annotations", SidecarInjectionSpec{RewriteAppHTTPProbe: false}, map[string]string{annotationRewriteAppProbers: "true"}, true},
```

Comment:
add another test case where annotationsMap is nil?

Reply:
added more test cases including nil case